### DISCUSSION TITLE
Is there any way to disable or turn off i18n base on a request ?!

### DISCUSSION THREAD

**Discussion by yazdanqm** [nuxt-modules/i18n#2462] [Q&A]
### Environment

export default

### Reproduction

```

``` 

### Describe the bug

I don't know this is a bug or not , but is there any way or feature to disable whole I18n based on an API request ?! i i used I18n as an Addon and my users can disable or enable it. i want to disable it and its routes and all configs dynamically, how can i do this in Nuxt config or out of it ?

**Comment by kazupon** [comment]
This is not issue, so question.
I'll move to github discussions.

**Comment by kazupon** [comment]
We need more context your use-case situation. 🙏

**Comment by yazdanqm** [comment]
> 'll move to github discussions.



> We need more context your use-case situation. 🙏

Hi and thank you for your response
actually we have a shop builder that has a admin panel and a lot of template, we want to have a addon or plugin that makes our templates multi-language. 
I created it with nuxt-i18n and everything works fine but the problem is when the admin doesn't want multi-language template and turns it off, or he can select languages he want. 
the main problem is all routes generates based on my locales array, how can i turn off this i18n config based on a async/await request ? or how can i get locales from backend? is there any solution for this ?
